# Assignment 2

To predict the price of a house using different parameters ANN Regression Models

## Making Necessary Imports

In [56]:
import numpy as np # Matrix Algebra
import pandas as pd # To load data

#Preprocessing
from sklearn.preprocessing import StandardScaler 

# Train vs Test
from sklearn.model_selection import train_test_split 

from sklearn.metrics import r2_score # Performance Metric
from tensorflow import keras # Neural Network Architecture

## Loading the Dataset

In [57]:
from google.colab import drive
drive.mount('/content/drive')
df = pd.read_csv("/content/drive/MyDrive/colabdata/House Price India.csv")
df.info()

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 23 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     14620 non-null  int64  
 1   Date                                   14620 non-null  int64  
 2   number of bedrooms                     14620 non-null  int64  
 3   number of bathrooms                    14620 non-null  float64
 4   living area                            14620 non-null  int64  
 5   lot area                               14620 non-null  int64  
 6   number of floors                       14620 non-null  float64
 7   waterfront present                     14620 non-null  int64  
 8   number of views                        14620 non-null  int64  
 9   condition

In [58]:
feature_list = [
    "id",
    "Date",
    "number of bedrooms",
    "number of bathrooms",
    "living area",
    "lot area",
    "number of floors",
    "waterfront present",
    "number of views",
    "condition of the house",
    "grade of the house",
    "Area of the house(excluding basement)",
    "Area of the basement",
    "Built Year",
    "Renovation Year",
    "Postal Code",
    "Lattitude",
    "Longitude",
    "Price",
]
data = df[feature_list]
X = data.drop(['Price'],axis=1)
y = data['Price']
X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=True)

## Preprocessing

In [59]:
scaler = StandardScaler()
scaler.fit(X_train)
X_train = scaler.transform(X_train)
X_test = scaler.transform(X_test)

## Neural Network Model Architecture

In [102]:
model = keras.Sequential(
    [
        keras.layers.Dense(1024, activation="relu"),
        keras.layers.Dense(512, activation="elu"),
        keras.layers.Dense(512, activation="relu"),
        keras.layers.Dense(512, activation="softplus"),
        keras.layers.Dense(512, activation="relu"),
        keras.layers.Dense(1),
    ]
)

model.compile(optimizer="adam", loss="mse", metrics=["mae"])
model.fit(X_train, y_train)

343/343 [==============================] - 11s 26ms/step - loss: 80313950208.0000 - mae: 147581.5469


## Make Test Data Predictions

In [103]:
pred = model.predict(X_test)
pred = np.array([i for i, in pred])
r2_score(y_test, pred, sample_weight=None)

115/115 [==============================] - 1s 5ms/step


0.8504556282367239

### Highest Possible R2 Score = 1.0
Our Score = 0.85 which is ideal 